src/ucp/wireup/address.c

```diff
@@ -1423,6 +1423,35 @@ ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep, void *buffer, size_t size,
     return UCS_OK;
 }
 
+ucs_status_t ucp_address_length(ucp_worker_h worker, ucp_ep_h ep,
```

Comment:
IMO better pass `const ucp_ep_config_key_t*` instead of ep to this function, to show that we don't need an actual endpoint to calculate the address size, but only the configuration info.
This way maybe can revert some of the changes in this PR because no need to create endpoint anymore to check if address fits the CM private data

Reply:
I'll pass ```ucp_ep_config(ep)```(i.e. ```ucp_ep_config_t```) here.
Reason:
1) After adding ```ucp_cm_ep_priv_space_is_enough``` function, both server and client need to do the same check. In both server & client, they could get ```ucp_ep_config_t``` directly.
2) After adding ```ucp_cm_ep_priv_space_is_enough``` function both in server and client side, ```ucp_cm_ep_priv_data_pack``` will only use ```ucs_error``` to report error if the cm private space is not enough.

Reply:
Done